modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -411,4 +558,42 @@ public Query getUnsupportedQuery() {
         }
     }
 
+    static class Range {
+
+        final String fieldName;
+        final byte[] lowerPoint;
+        final byte[] upperPoint;
+        final int length;
+
+        Range(String fieldName, byte[] lowerPoint, byte[] upperPoint, Number decodedWidth) {
+            this.fieldName = fieldName;
+            this.lowerPoint = lowerPoint;
+            this.upperPoint = upperPoint;
+            this.length = decodedWidth.intValue();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Range range = (Range) o;
+            return Objects.equals(fieldName, range.fieldName) &&
+                Arrays.equals(lowerPoint, range.lowerPoint) &&
+                Arrays.equals(upperPoint, range.upperPoint);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(fieldName, lowerPoint, upperPoint);
```

Comment:
I'm not sure whether this is correct since the hashcode of byte[] is the identity hashcode by default? Unless Objects.hash is aware of this?